server.cpp

```diff
@@ -85,17 +107,50 @@ int main(int argc, char* argv[])
         tcp::resolver resolver(io_service);
         tcp::acceptor acceptor(io_service, *resolver.resolve(tcp::resolver::query(host, port)));
 
+        char buff[1024];
+
+        std::ofstream fout;
+        if (outfile.empty())
+            fout.close();
+
         for (;;)
         {
             tcp::socket socket(io_service);
             acceptor.accept(socket);
 
-            const std::string message = make_daytime_string();
+            int bytes = read(socket, boost::asio::buffer(buff), boost::bind(read_complete, buff, _1, _2));
+
+            std::string msg(buff, bytes);
+
+            size_t str_end_pos = msg.find("\r");
+            std::string start_str = msg.substr(0, str_end_pos);
+            std::string http_version = start_str.substr(str_end_pos - 3);
```

Comment:
1. `const`?
2. Version is not only the last 3 symbols, it's the whole "HTTP/1.1" thing.